### PR TITLE
Fix formatting of negative values

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix formatting of negate/negative values (Before this patch `-(-1)` was formatted to `--1` which is not valid)

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Fuzzer said that the formatting of `SELECT -(1)` is inconsistent.

**Comment by alexey-milovidov** [comment]
This PR is only relevant to fuzzer. No issues with parsing of any possible user input.
```
:) SELECT -(-1)

SELECT 1
```

**Comment by alexey-milovidov** [comment]
Let's continue.

**Comment by alexey-milovidov** [comment]
This fix is very appreciated.

File: src/Parsers/ASTFunction.cpp [hunk 1]
```diff
@@ -229,68 +229,32 @@ void ASTFunction::formatImplWithoutAlias(const FormatSettings & settings, Format
                 }
 
                 const auto * literal = arguments->children[0]->as<ASTLiteral>();
-                /* A particularly stupid case. If we have a unary minus before
```
Comment (azat) [inline src/Parsers/ASTFunction.cpp:232]: So I had to remove this, otherwise it makes formatting of negate inconsistent, i.e. `-(-(-1))` -> ` -(1)` -> `-1`

**Comment by azat** [comment]
>Conflicting files
>src/Parsers/ASTFunction.cpp

Rebased.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Awesome!

**Comment by alexey-milovidov** [comment]
We also have a similar case with `NOT NOT x`

**Comment by azat** [comment]
>We also have a similar case with NOT NOT x

Yep, saw this too, will take a look later.

**Pull request merged**